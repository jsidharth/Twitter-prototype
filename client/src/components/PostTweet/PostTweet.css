.cardContainer {
  width: 600px;
}

.postTweetSeparator {
  background-color: #E6ECF0;
  height: 10px;
}

.inputStyle {
  display: none;
  float: right;
}

.cardWidth {
  width: 600px;
}

.paperHeight {
  padding-top: 10px;
  height: 50px;
  font-weight: 800;
  font-size: 19px;
  font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Ubuntu,
    'Helvetica Neue', sans-serif;
  padding-left: 15px;
  border-style: solid;
  border-width: 0.5px;
  border-color: #d0d4d6;
}

.cardContent {
  font-size: inherit;
  margin-bottom: 0px;
}

.tweetImagePost {
  object-fit: cover;
  margin-top: 20px;
  margin-bottom: 10px;
  width: 100%;
  height: 250px;
}

.flexImageTweet {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

.profileImageTweet {
  margin: 0 auto;
  max-width: 50px;
  max-height: 50px;
  border-radius: 50%;
}

.tweetText {
  margin-left: 20px;
  margin-top: 0px;
  width: 95%;
  height: 50px;
  outline: none;
}

.autoExpandDiv {
  display: inline-block;
  min-height: 70px;
  width: 95%;
  margin-bottom: 10px;
}

.textArea {
  margin-left: 20px;
  margin-top: 0px;
  width: 97%;
  height: 100%;
  outline: none;
  margin-top: 0px;
  overflow: auto;
  resize: none;
  border: none;
  font-size: 19px;
  font-weight: 400;
  overflow-wrap: break-word;
}

div[data-placeholder]:empty:not(:focus):not([data-div-placeholder-content]):before {
  content: attr(data-placeholder);
  float: left;
  margin-left: 2px;
  color: #b3b3b3;
}

.iconUpload {
  margin-left: 80px;
  display: 'none';
}

.postTweetBtn {
  background-color: #1da1f2;
  color: white;
  text-align: center;
  font-weight: 700;
  float: right;
  margin-top: 5px;
  padding-left: 15px;
  padding-right: 15px;
  padding-top: 7px;
  padding-bottom: 7px;
  border-radius: 20px;
  font-size: 15px;
  outline: none;
}

.flexUploadTweet {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
}

.flexIconCharsCount {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
}

.countMessageStyle {
  font-size: 17px;
  font-weight: 400;
  color: #b3b3b3;
  margin-left: 5px;
  margin-top: 1px;
}

*:focus {
  outline: 0 !important;
}