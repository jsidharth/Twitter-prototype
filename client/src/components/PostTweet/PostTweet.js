import React, { Component } from 'react';
import './PostTweet.css';
import Image from 'react-bootstrap/Image';
import { Card, CardContent, Paper, Icon } from '@material-ui/core';
import { InsertPhoto } from '@material-ui/icons';
import { connect } from 'react-redux';
import { imageActions, tweetActions } from '../../js/actions/index';

class PostTweet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tweetText: '',
      pictures: [],
    };

    this.tweetTextHandler = this.tweetTextHandler.bind(this);
    this.uploadImage = this.uploadImage.bind(this);
    this.postTweet = this.postTweet.bind(this);
  }

  tweetTextHandler = e => {
    this.setState({
      tweetText: e.target.value,
    });
  };

  uploadImage = e => {
    const data = new FormData();
    if (this.upladTweetImage.files && this.upladTweetImage.files.length) {
      data.append('file', this.upladTweetImage.files[0] || '');
      this.props.upload(data);
    }
  };

  postTweet = () => {
    // change userId

    if (this.state.tweetText.length > 280) {
      console.log('max length exceeded');
    } else {
      const data = {
        userId: '5dcb31841c9d440000b0d332',
        tweetText: this.state.tweetText,
        imageUrl: this.props.imageUrl,
      };
      console.log(data);
      this.props.postTweet(data);
    }
  };

  render() {
    var count = 280 - this.state.tweetText.length;

    return (
      <Card className="cardWidth">
        <Paper className="paperHeight">Home</Paper>
        <CardContent className="cardContent">
          <div className="flexImageTweet">
            <div>
              {/* Include user profile image if available */}
              <Image
                src="/images/default_profile_bigger.png"
                roundedCircle
                className="profileImage"
              />
            </div>

            <div className="autoExpandDiv">
              <textarea
                className="textArea"
                onChange={this.tweetTextHandler}
                placeholder="What's happening?"
              ></textarea>
            </div>
          </div>

          {this.props.imageUrl ? (
            <img src={this.props.imageUrl} className="tweetImage" alt="Tweet Image" />
          ) : null}
          <div className="flexUploadTweet">
            <div className="flexIconCharsCount">
              <div className="iconUpload">
                <input
                  accept="image/*"
                  id="icon-button-file"
                  type="file"
                  onChange={this.uploadImage}
                  ref={ref => {
                    this.upladTweetImage = ref;
                  }}
                />
                <label htmlFor="icon-button-file">
                  <Icon color="primary">
                    <InsertPhoto />
                  </Icon>
                </label>
              </div>
              <div className="countStyle">{`${count} characters remaining`}</div>
            </div>
            <button className="postTweetBtn" onClick={this.postTweet}>
              Tweet
            </button>
            <input
              type="text"
              name="tweetText"
              onChange={this.tweetTextHandler}
              required
              maxlength="280"
            />
          </div>
        </CardContent>
      </Card>
    );
  }
}

const mapStateToProps = state => {
  return {
    imageUrl: state.image.imageUrl,
    tweetPostedFlag: state.tweet.tweetPostedFlag,
  };
};

const mapDispatchToProps = dispatch => ({
  upload: data => dispatch(imageActions.upload(data)),
  postTweet: data => dispatch(tweetActions.postTweet(data)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PostTweet);
