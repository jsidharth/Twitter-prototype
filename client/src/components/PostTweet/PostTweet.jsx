/* eslint-disable jsx-a11y/label-has-associated-control */
/* eslint-disable no-underscore-dangle */
/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import './PostTweet.css';
import { connect } from 'react-redux';
import { AiOutlinePicture } from 'react-icons/ai';
import { imageActions, tweetActions } from '../../js/actions/index';

class PostTweet extends Component {
  constructor(props) {
    super(props);
    this.state = {
      tweetText: '',
    };

    this.tweetTextHandler = this.tweetTextHandler.bind(this);
    this.uploadImage = this.uploadImage.bind(this);
    this.postTweet = this.postTweet.bind(this);
  }

  tweetTextHandler = e => {
    this.setState({
      tweetText: e.target.value,
    });
  };

  uploadImage = () => {
    const data = new FormData();
    const { upload } = this.props;
    if (this.uploadTweetImage.files && this.uploadTweetImage.files.length) {
      data.append('file', this.uploadTweetImage.files[0] || '');
      upload(data);
    }
  };

  postTweet = () => {
    const { tweetText } = this.state;
    const { postTweet, userId, imageUrl, fetchUpdatedFeed  } = this.props;
    if (tweetText.length > 280) {
      console.log('max length exceeded');
    } else {
      const data = {
        userId,
        tweetText,
        imageUrl,
      };
      postTweet(data).then(() => {
        const fetchFeedPayload = {
          userId,
          count: 10,
          offset: 0,
        };
        fetchUpdatedFeed(fetchFeedPayload);
        this.setState({
          tweetText: '',
        });
      });
    }
  };

  render() {
    const { tweetText } = this.state;
    const { imageUrl, user } = this.props;
    const count = 280 - tweetText.length;

    return (
      <div className="cardContainer">
        <div className="cardWidth">
          <div className="paperHeight">Home</div>
          <div className="cardContent">
            <div className="flexImageTweet">
              <div>
                <img
                  src={user.profilePic ? user.profilePic : '/images/default_profile_bigger.png'}
                  className="profileImageTweet"
                  alt="User profile"
                />
              </div>
              <div className="autoExpandDiv">
                <textarea
                  className="textArea"
                  onChange={this.tweetTextHandler}
                  placeholder="What's happening?"
                  maxLength="280"
                  value={tweetText}
                />
              </div>
            </div>
            {imageUrl ? <img src={imageUrl} className="tweetImagePost" alt="Tweet" /> : null}
            <div className="flexUploadTweet">
              <div className="flexIconCharsCount">
                <div className="iconUpload">
                  <input
                    className="inputStyle"
                    accept="image/*"
                    id="icon-button-file"
                    type="file"
                    onChange={this.uploadImage}
                    ref={ref => {
                      this.uploadTweetImage = ref;
                    }}
                  />
                  <label htmlFor="icon-button-file">
                    <AiOutlinePicture color="#1da1f2" size={20} />
                  </label>
                </div>
                <div className="countMessageStyle">{`${count} characters remaining`}</div>
              </div>
              <button type="button" className="postTweetBtn" onClick={this.postTweet}>
                Tweet
              </button>
            </div>
          </div>
        </div>
        <div className="postTweetSeparator" />
      </div>
    );
  }
}

const mapStateToProps = state => ({
  imageUrl: state.image.imageUrl,
  tweetPostedFlag: state.tweet.tweetPostedFlag,
  userId: state.user.currentUser._id,
  user: state.user.currentUser,
});

const mapDispatchToProps = dispatch => ({
  upload: data => dispatch(imageActions.upload(data)),
  postTweet: data => dispatch(tweetActions.postTweet(data)),
  fetchUpdatedFeed: data => dispatch(tweetActions.fetchUpdatedFeed(data)),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(PostTweet);
