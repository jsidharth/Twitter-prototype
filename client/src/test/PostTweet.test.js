// Link.react.test.js
import React from 'react';
import renderer from 'react-test-renderer';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import PostTweet from '../components/PostTweet/PostTweet';

configure({ adapter: new Adapter() });

it('render correctly the Post Tweet component', () => {
  const postTweetComponent = renderer.create(<PostTweet />).toJSON();
  expect(postTweetComponent).toMatchSnapshot();
});
