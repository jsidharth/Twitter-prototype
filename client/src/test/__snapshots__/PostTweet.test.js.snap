// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly the Post Tweet component 1`] = `
<div
  className="cardContainer"
>
  <div
    className="cardWidth"
  >
    <div
      className="paperHeight"
    >
      Home
    </div>
    <div
      className="cardContent"
    >
      <div
        className="flexImageTweet"
      >
        <div>
          <img
            alt="User profile"
            className="profileImageTweet"
            src="/images/default_profile_bigger.png"
          />
        </div>
        <div
          className="autoExpandDiv"
        >
          <textarea
            className="textArea"
            maxLength="280"
            onChange={[Function]}
            placeholder="What's happening?"
            value=""
          />
        </div>
      </div>
      <div
        className="flexUploadTweet"
      >
        <div
          className="flexIconCharsCount"
        >
          <div
            className="iconUpload"
          >
            <input
              accept="image/*"
              className="inputStyle"
              id="icon-button-file"
              onChange={[Function]}
              type="file"
            />
            <label
              htmlFor="icon-button-file"
            >
              <svg
                color="#1da1f2"
                fill="currentColor"
                height={20}
                size={20}
                stroke="currentColor"
                strokeWidth="0"
                style={
                  Object {
                    "color": "#1da1f2",
                  }
                }
                viewBox="0 0 1024 1024"
                width={20}
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 632H136v-39.9l138.5-164.3 150.1 178L658.1 489 888 761.6V792zm0-129.8L664.2 396.8c-3.2-3.8-9-3.8-12.2 0L424.6 666.4l-144-170.7c-3.2-3.8-9-3.8-12.2 0L136 652.7V232h752v430.2zM304 456a88 88 0 1 0 0-176 88 88 0 0 0 0 176zm0-116c15.5 0 28 12.5 28 28s-12.5 28-28 28-28-12.5-28-28 12.5-28 28-28z"
                />
              </svg>
            </label>
          </div>
          <div
            className="countMessageStyle"
          >
            280 characters remaining
          </div>
        </div>
        <button
          className="disabledPostTweetBtn"
          disabled={true}
          onClick={[Function]}
          type="button"
        >
          Tweet
        </button>
      </div>
    </div>
  </div>
  <div
    className="postTweetSeparator"
  />
</div>
`;
